### PR TITLE
Support "est" stats in total/mean search combiners

### LINKED ISSUE #7195: Use stats wildcards (highest, secondhighest, etc...) in average searches
### Proposed change

Ability to use stats wildcards `highest`, `secondhighest` etc...  for average searches, such as `basestat:highest&secondhighest:>=16`. This is currently not possible, as confirmed in #6819. I couldn't find an opened issue so I assumed this fell under low priority, or maybe is kind of tricky to implement.

### How does this fit into your workflow?

Armor that rolls with 2 or 3 very high stats are usually the most useful for building loadouts with 2x100 stats. This feature would enable filtering for these rolls without assuming which combination of stats are the highest, avoiding filling DIM filters with stats permutations.

In practical terms, this:
```
basestat:mobility&recovery:>=17.5 or
basestat:mobility&discipline:>=17.5 or
basestat:mobility&intellect:>=17.5 or
basestat:mobility&strength:>=17.5 or
basestat:resilience&recovery:>=17.5 or
basestat:resilience&discipline:>=17.5 or
basestat:resilience&intellect:>=17.5 or
basestat:resilience&strength:>=17.5 or
basestat:recovery&discipline:>=17.5 or
basestat:recovery&intellect:>=17.5 or
basestat:recovery&strength:>=17.5 or
basestat:discipline&intellect:>=17.5 or
basestat:discipline&strength:>=17.5 or
basestat:intellect&strength:>=17.5
```

would become this:
```
basestat:highest&secondhighest:>=17.5
```

### PR DESCRIPTION
**Pull request by bhollis**
Fixes #7195

### CODE REVIEW

**Review by nev-r — APPROVED** [review]
seeeeems working well. good for measuring "a pair of stats adds up to X".
image

File: config/i18n.json [hunk 1]
```diff
@@ -246,8 +246,9 @@
     "Stackable": "Shows items that can stack (ammo synths, strange coin, etc)",
     "StatLower": "Shows armor whose stats are strictly lower than another of the same type of armor.",
     "CustomStatLower": "Shows armor whose stats are strictly lower than another of the same type of armor, only taking into account stats in that class' custom stat total list.",
-    "Stats": "Shows items based on a specific stat value. Supports stat addition by connecting multiple stat names with the + symbol.",
-    "StatsBase": "Filters armor based on its base stat value, not including attached mods or masterworking. Supports stat addition by connecting multiple stat names with the + symbol.",
+    "Stats": "Shows items based on a specific stat value. $t(Filter.StatsExtras)",
+    "StatsBase": "Filters armor based on its base stat value, not including attached mods or masterworking. $t(Filter.StatsExtras)",
+    "StatsExtras": "Supports stat addition by connecting multiple stat names with the + symbol.  There are also special keywords highest, secondhighest, thirdhighest, etc. which match stats based on their rank within an item's stats.",
```
Comment (delphiactual) [inline config/i18n.json:251]: this also seems to work with the `&` symbol

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 * However, we have managed to fix DIM so it doesn't crash loop in the Steam overlay. Until the next time Steam updates...
 * Loadout Optimizer performance has been improved significantly - so much so that we now always look at all possible combinations of armor. Previously we trimmed some items out to get below an a number that we could process in time. This means that your LO builds are now guaranteed to be optimal, and the "Max" range shown in the stat tiles will always be accurate.
 * We no longer cap stats in the Loadout Optimizer's tooltips so you can see how far over 100 a stat goes.
+* Stat searches support keywords like "highest" and "secondhighest" in stat total/mean expressions. e.g. basestat:highest&secondhighest:>=17.5
```
Comment (delphiactual) [inline docs/CHANGELOG.md:7]: as stated here

File: config/i18n.json (hunk 1 above)
Comment (delphiactual) [inline config/i18n.json:251]:
```suggestion
    "StatsExtras": "Supports stat addition by connecting multiple stat names with the + or & symbol.  There are also special keywords highest, secondhighest, thirdhighest, etc. which match stats based on their rank within an item's stats.",
```

**Pull request merged**